### PR TITLE
content_view_filter_rule module

### PR DESCRIPTION
**Pull request by evgeni**
split out of #1547

### CODE REVIEW

File: plugins/modules/content_view_filter_rule.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+            if not module.desired_absent:
+                # if the state is present and the module_id is NOT in the exising list, add module_stream_id.
+                if not existing_rule:
+                    module.foreman_params['module_stream_ids'].append(module_stream['id'])
+
+                # if the state is present and the module_id IS in the list,
+                # make sure that the current and desired state are identical
+                else:
+                    content_view_filter_rule = module.foreman_params
```
Comment (evgeni) [inline plugins/modules/content_view_filter_rule.py]: This *shouldn't* be necessary (but it is), so worth investigating further.

File: plugins/modules/content_view_filter_rule.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+        elif filter_type == 'package_group':
+            # this filter type support many rules
+            # the name is the key to finding the proper one and is required for these types
+            # uuid is also a required value creating, but is implementation specific and not easily knowable to the end user - we find it for them
+            content_view_filter_rule = module.find_resource_by_name('content_view_filter_rules', module.foreman_params['name'],
+                                                                    params=search_scope, failsafe=True)
+
+            package_group = module.find_resource_by_name('package_groups', module.foreman_params['name'], params=cv_scope)
+            module.foreman_params['uuid'] = package_group['uuid']
```
Comment (evgeni) [inline plugins/modules/content_view_filter_rule.py]: I'm wondering if simplifying this to the following will make it better readable or not?

```suggestion
        elif filter_type in ('rpm', 'docker', 'package_group'):
            # these filter types support many rules
            # the name is the key to finding the proper one and is required for these types
            content_view_filter_rule = module.find_resource_by_name('content_view_filter_rules', module.foreman_params['name'],
                                                                    params=search_scope, failsafe=True)

            if filter_type == 'package_group':
                # uuid is also a required value creating, but is implementation specific and not easily knowable to the end user - we find it for them
                package_group = module.find_resource_by_name('package_groups', module.foreman_params['name'], params=cv_scope)
                module.foreman_params['uuid'] = package_group['uuid']
```

Comment (parmstro) [reply plugins/modules/content_view_filter_rule.py]: Yes. That works and means less maintenance.

**Review by parmstro — APPROVED** [review]
Yes, Evgeni. I think that the change is good. Clearer and less maintenance.

**Pull request merged**